RFCs/FS-1128-better-interpolated-triple-quoted-strings.md

```diff
@@ -0,0 +1,119 @@
+
+# Motivation
+
+Triple quoted strings are already quite handy for embedding e.g. XML in string literals, but not as convenient for embedding some other languages that use `{` and `}` in their syntax, e.g. JSON or CSS. Currently every occurrence of a curly brace in an interpolated string needs to be escaped, which can quickly become tedious. With this extended syntax, only the braces around interpolated expression need special attention, and that is where programmer's focus already is.
+
+The motivation is not to enable any novel use cases, but to make working with interpolated string literals a much more seamless experience in the use cases mentioned above.
+
+This change should:
+1. be backwards compatible with existing support for triple quoted strings
+2. align with C# raw string literals reasonably well
+
+# Detailed design
+
+The existing form of interpolated triple quoted strings: `$"""...{}..."""` is extended to allow multiple dollar signs at the begining of the string literal. The count of these initiatory `$` characters indicates how many `{` and `}` characters are used to delimit interpolation expression within the content of the literal.
+
+Important note: To maintain backward compatibility, there are no changes in semantics for a single-`$` case. This means that it is an edge case in terms of how this feature behaves. It would be nice to have a clean design that covers all cases, but it is not worth breaking backward compatibility for. Therefore the below description applies to the literals starting with 2 or more `$`.
+
+Triple quoted string literals starting with 2 or more `$` characters have no escaping mechanism for `{` or `}`. Instead, the literal can always be constructed in such a way as to ensure that interpolation delimiters will not collide with other curly braces in the content. In a literal that starts with `N` `$` characters, a sequence of consecutive `{` can be at most `2*N-1` characters long and for any sequence of more than `N`, the innermost `N` `{` characters delimit the interpolation, while the outter up to `N-1` are treated as content (analogously for `}`). A sequence of `2*N` or more `{` (or `}`) will result in a compilation error.
+
+Example:
+```
+let text = "3 curly braces delimit interpolation expression"
+$$$"""
+In a string literal with 3x$, { or {{ are simply treated as content
+whereas {{{text}}}.
+It is also allowed to have {, {{, } or }} adjacent to the interpolation delimiters as so:
+{{{{41+1}}}} = {42}
+"""
+```
+
+# Drawbacks
+
+Even though this only extends already existing syntax, it still can be considered yet another way of doing string interpolation. Moreover, it can't be fully aligned with analogous C# feature (raw string literals) and stay backward compatible at the same time, so this could be a source of confusion (but note that triple quoted string literals are already similar but not really aligned with C#'s raw string literals anyway).
```

Comment:
You mention “raw string literals” several times. Can we briefly explain them? Assume us fsharpers don’t know that feature (I don’t). How are they not aligned?

Reply:
There are links at the bottom of this RFC with description of raw string literals, I can refer to them here.

If you think there should be some direct explanation of raw string literals in this RFC and how they differ from triple quoted strings, I can think about that, but I'd argue it shouldn't go in this particular section anyway.